keras/layers/containers.py

```diff
@@ -277,6 +277,64 @@ def add_node(self, layer, name, input=None, inputs=[],
         if create_output:
             self.add_output(name, input=name)
 
+    def add_shared_node(self, layer, name, inputs=[], merge_mode=None, concat_axis=-1, dot_axes=-1, outputs=[], create_output=False):
```

Comment:
Please add a docstring detailing the meaning of each argument. It is non-obvious.